### PR TITLE
add TERT SV info to telomerase plot script

### PR DESCRIPTION
**Pull request by jharenza**
#### What scientific question is your analysis addressing?
This addresses part of the reviewer comment asking how telomerase scores are related to TERT rearrangements or mutations. My eyes must have glazed past "rearrangements" in the review, so here, I am adding the TERT SVs for assessment. 

#### What was your approach?
Since SVs are often difficult to assess pathogenicity for, I took a very conservative approach here. I found [knotAnnotSV](https://academic.oup.com/nar/article/49/W1/W21/6281473), a companion tool to AnnotSV, which helps asses function of SVs, including ACMG pathogenicity call. To use this, I had to create a BED file from manta SV, plug it into https://lbgi.fr/AnnotSV, then assessed the calls for ACMG == 4|5, which correspond to likely pathogenic (LP) or pathogenic (P). 

#### What GitHub issue does your pull request address?
NA, revision

#### Which areas should receive a particularly close look?
NA

#### Is there anything that you want to discuss further?
Note: I reran the whole module, but realized later I maybe should have just run script 06, which, oddly wasn't in the shell script so I added it. Not sure if that was not added on purpose.

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes

#### What is your summary of the results?
There were 2 samples with LP TERT SVs (of 3 with a TERT SV). There were also multiple SVs overlapping TERT or subsuming TERT (eg large arm deletions), and although knotAnnotSV often annotated these full-length SVs as P/LP, I do not think we can confidently say these are "TERT" SVs unless a breakpoint is actually in TERT. So, I collected these in the BED file, but did not associate them with being TERT-driven.

The 2 samples with LP TERT SVs correspond to two sample ids from 2 patients. The corresponding RNAs are both polyA, so I did not end up plotting / combining these in any way with the TERTp mutations. 

In summary, while the EXTEND paper shows a nice plot of higher scores in TERT-rearranged tumors, we don't have the power to assess that here and we can refer back to the paper for  association.

I really wanted to just cover bases and check SVs here and note this in the script.

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by sjspielman** [comment]
I know this PR is still in progress since a table needs to be added, as discussed on Slack, but wanted to leave a note to that end @jharenza  - 
I see you've fixed the not-so-great bug where `06-TERTp-EXTENDScores.R` never got added into `RUN-telomerase-activity-prediction.sh` here: https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/ae299d9e1ff1c270e5198d8aa0bdff4f1fad1727/analyses/telomerase-activity-prediction/RUN-telomerase-activity-prediction.sh#L54, so that’s good!

But! The way that the module script logic is written, `06-TERTp-EXTENDScores.R` will not get run as part of the manuscript run - even if there were no bug here, `06-TERTp-EXTENDScores.R` would not have been run anyways over in #1672. So, if we want to make a table that is used in the MS, we have to make a choice here. We can either:
- We keep the logic in `RUN-telomerase-activity-prediction.sh` as is, and actually make that table over in `tables/`.
- Move the code that runs  `06-TERTp-EXTENDScores.R` _out of_ the `if/else` logic entirely so that it always runs when the module is run. But, as I write this, thinking we should move this line out of the if/else either way.

**Comment by jharenza** [comment]
Hi @sjspielman sorry for the delay. I updated this, and moved the 06 script out of if/else. Ready for review!

File: analyses/telomerase-activity-prediction/06-TERTp-EXTENDScores.R [hunk 1]
```diff
@@ -59,6 +77,39 @@ metadata_df <- read_tsv(metadata_file, guess_max = 10000)
...
+# Read in and export TERT SV data in BED format for knotAnnotSV to assess pathogenicity of the SV call
+# This BED file was put into https://lbgi.fr/AnnotSV/ and the TSV downloaded for ACMG assessment below
+sv_df <- read_tsv(sv_file) %>%
+  # only keep PASS variants
+  filter(FILTER == "PASS",
+         # let's check all 
+         grepl("TERT", Gene.name)) %>%
+  select(chrom = SV.chrom, chromStart = SV.start, chromEnd = SV.end, name = AnnotSV.ID, 
+                  SVtype = SV.type, Sample = Kids.First.Biospecimen.ID.Tumor) %>%
+  write_tsv(file.path(results_dir, "mantaSV_tert.bed"))
```
Comment (sjspielman) [inline analyses/telomerase-activity-prediction/06-TERTp-EXTENDScores.R:89]: Can we move this and subsequent file definitions to top of this R script so they aren't hardcoded within `write_tsv` lines?

**Review by sjspielman — COMMENTED** [review]
For a first round of review, I want to mostly clarify what exactly we're doing in this PR. For this analysis, I do expect to see an output table, which we have in `extend_scores_tert_alteration.tsv`. But, I do not understand where `input/AnnotSV_VJFSh10MHo.tsv` came from and how/why it's being used, as there's no docs for it. Was it generated from a new tool `knotAnnotSV` as you described in the PR? It appears that that is a web server (?), which means this file is not reproducible. Any new tools need to be part of `Docker` and full documented in our Key Resources table for the MS, and this will involve a bit of a lift. So, more generally, I will ask: Why do we need such a tool at all? I don't really see a need to assess _pathogenicity_ of any SVs in the first place, as this wasn't mentioned in the reviewer comment. I guess I'm not sure why we can't just list any rearrangements, and ignore their potential consequences.

Something might be up with the output table as well, as I count 8 rows for TERTp present but our plot only has 6 points for stranded.. can you check the numbers there?
https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/e00ecb0e35bc54146a35c59fce71cf19a31df6c1/analyses/telomerase-activity-prediction/06-TERTp-EXTENDScores.R#L161

I'll note also that some of these diffs here (such as `analyses/telomerase-activity-prediction/results/EXTENDScores_Ependymal_tumor.tsv`) should go away when the current version of `master` is merged in, which I will leave to you!

File: analyses/telomerase-activity-prediction/06-TERTp-EXTENDScores.R (hunk 1 above)
Comment (jharenza) [reply analyses/telomerase-activity-prediction/06-TERTp-EXTENDScores.R:89]: I am removing this now

**Comment by jharenza** [comment]
>Something might be up with the output table as well, as I count 8 rows for TERTp present but our plot only has 6 points for stranded.. can you check the numbers there?

This table has _all_ variants with a column for `RNA_library` so I can mention all in review.

**Comment by jharenza** [comment]
@sjspielman I think this is ready again - For `analyses/telomerase-activity-prediction/results/EXTENDScores_Ependymal_tumor.tsv`, this was updated with a rerun, didn't change with merge from master - do you want me to start a fresh rerun here?

**Comment by sjspielman** [comment]
> For `analyses/telomerase-activity-prediction/results/EXTENDScores_Ependymal_tumor.tsv`, this was updated with a rerun, didn't change with merge from master - do you want me to start a fresh rerun here?

I know what you have there is correct, so leave as is! I know I came across this updated result at some point and maybe I messed up committing it..

**Review by sjspielman — APPROVED** [review]
This seems pretty set to me, and I see the differences in numbers (6 vs 8) I saw were polya vs. stranded. I left a couple comments about code comments but nothing is major, so I will go ahead and approve now assuming that's all taken care of! 

So, in the end, is the plan to point to this table in the response to reviewers? We can also add a quick sentence about how many tumors have SVs as needed?

File: analyses/telomerase-activity-prediction/06-TERTp-EXTENDScores.R [hunk 2]
```diff
@@ -59,6 +77,33 @@ metadata_df <- read_tsv(metadata_file, guess_max = 10000)
 maf_df <- read_tsv(maf_file)
 hotspot_df <- read_tsv(hotspot_file)
 
+# How many samples have TERT SVs and how many are pathogenic/likely pathogenic?
```
[the remark this answers is not in the document]
Comment (jharenza) [reply analyses/telomerase-activity-prediction/06-TERTp-EXTENDScores.R:80]: >So, in the end, is the plan to point to this table in the response to reviewers? We can also add a quick sentence about how many tumors have SVs as needed?

Yup, already incorporated it earlier!

File: analyses/telomerase-activity-prediction/06-TERTp-EXTENDScores.R [hunk 3]
```diff
@@ -59,6 +77,33 @@ metadata_df <- read_tsv(metadata_file, guess_max = 10000)
...
+  pull(sample_id) %>%
+  unique()
+  
+# Now get the RNASeq stranded IDs for those sample ids
+# note that not all of these may be stranded!
+rna_tertsv_bs_ids <- metadata_df %>%
+  filter(sample_id %in% tertsv_sample_ids, 
+         experimental_strategy == "RNA-Seq") %>%
+  select(Kids_First_Biospecimen_ID, RNA_library) 
+# One polyA, other WGS only. We we won't add polyA to stranded plot.
```
[the remark this answers is not in the document]
Comment (jharenza) [reply analyses/telomerase-activity-prediction/06-TERTp-EXTENDScores.R:109]: ah this was a bit leftover from using the website annots. I think this was referring to 7316-3240 which doesn't have an RNA sample as being WGS only. Then, 1/2 of the ones with RNA was polyA. I will clarify.
